behind-the-veil-siteroot/imports/ui/components/DatabaseHelper.jsx

```diff
@@ -86,19 +84,243 @@ export function getServices(service_publication, params, filter, requireArtist =
     return [isLoading, servicesData]
 }
 
-// many bookings
+/**
+ * Used for to get a list of bookings data (includes their image and service data automatically)
+ *
+ * @param booking_publication - name of the publication to get bookings from (determines which subset of booking is required)
+ * @param params - the parameters used for that booking publication, such as username, etc.
+ * @param filter - the filter used to filter for the specific subset of booking (from the subscribed publication) this is an
+ * object with key=booking attribute name, value=filter value; e.g. { username: "abcd123" }
+ * @returns {[boolean, Object]} - returns a boolean isLoading that is true when loading, false when finished loading. The object
+ * contains the bookingData joined with its serviceData and serviceImage
+ */
 export function getBookings(booking_publication, params, filter) {
 
+    // get bookings from database
+    const isLoadingBooking = useSubscribe(booking_publication, ...params);
+    let bookingsData = useTracker(() => {
+        return BookingCollection.find(filter).fetch();
+    });
+
+    // get services from database
+    const isLoadingService = useSubscribe('all_services');
+    let servicesData = useTracker(() => {
+        return ServiceCollection.find().fetch();
+    });
+
+    // get service images from database
+    const isLoadingServiceImage = useSubscribe('service_images');
+    let imagesData = useTracker(() => {
+        return ImageCollection.find({"imageType": "service"}).fetch();
+    });
+
+    // variable for loading
+    const isLoading = isLoadingBooking() || isLoadingService() || isLoadingServiceImage();
+
+    // manual aggregation into bookingsData with its services and images
+    for (let i = 0; i < bookingsData.length; i++) {
+
+        // aggregate with service first
+        for (let j = 0; j < servicesData.length; j++) {
+            // find matching service ID
+            if (bookingsData[i].serviceId === servicesData[j]._id) {
+                bookingsData[i].serviceName = servicesData[j].serviceName;
+                bookingsData[i].serviceDesc = servicesData[j].serviceDesc;
+                break;
+            }
+        }
+        // then aggregate with the FIRST service image (cover)
+        for (let j = 0; j < imagesData.length; j++) {
+            // find matching image for the service
+            if (imagesData[j].imageType === "service" && bookingsData[i].serviceId === imagesData[j].target_id) {
+                bookingsData[i].serviceImageData = imagesData[j].imageData;
+                break;
+            }
+        }
+    }
+
+    console.log("bookingsData", bookingsData)
```

Comment:
Just wondering, are the console logs included in this PR included on purpose?

Reply:
removed all console logs